### PR TITLE
refactor(connector): directly deserialize config from map

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Avoid first converting a map into a json value, then deserialize it back.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14775?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #14768** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14768?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #14775** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14775?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/risingwavelabs/risingwave/14775?utm_source=stack-comment).

**Review by BugenZhao — COMMENTED** [review]
Oops. Seems not working as expected. 😕

**Comment by xiangjinwu** [comment]
```
e2e_test/source/basic/ddl.slt                                .. [FAILED]
failed to run `e2e_test/source/basic/ddl.slt`
Caused by:
    statement is expected to fail with error:
    	(multiline) db error: ERROR: Failed to run the query
    
    Caused by these errors (recent errors listed first):
      1: gRPC request to meta service failed: Internal error
      2: failed to create source worker
      3: missing field `properties.bootstrap.server`
    but got error:
    	db error: ERROR: Failed to run the query
    
    Caused by these errors (recent errors listed first):
      1: gRPC request to meta service failed: Internal error
      2: failed to create source worker
      3: Serde("missing field `properties.bootstrap.server`") at character 0
    
    [SQL] create source s(x int) WITH(unknown_field='1', connector='kafka') FORMAT PLAIN ENCODE JSON;
    at e2e_test/source/basic/ddl.slt:23
```
Seems only an appearance issue.

File: src/connector/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /// Deserialize from a `&HashMap` or `&BTreeMap` of string-like keys and values.
+    pub fn deserialize_from_map<'a, M, K, V>(map: &'a M) -> Result<Self, simd_json::Error>
+    where
+        &'a M: IntoIterator<Item = (&'a K, &'a V)>,
+        K: AsRef<str> + 'a,
+        V: AsRef<str> + 'a,
+    {
+        // The deserializing itself has nothing to do with `simd_json`. We just use it to
+        // provide a correct implementation of `deserialize_option` for optional fields.
```
Comment (xiangjinwu) [inline src/connector/src/utils.rs:30]: Does not quite get this part. Is there a unit test to demonstrate what would happen if it was simply `(k.as_ref(), v.as_ref())`? It is also confusing for caller to get a `simd_json::Error`.

Comment (xxchan) [reply src/connector/src/utils.rs:30]: Maybe return `anyhow::Error` here and thus save some conversions outside.

Comment (BugenZhao) [reply src/connector/src/utils.rs:30]: It's kind of counterintuitive, but actually we need something like this:

https://github.com/BugenZhao/rust-playground/blob/7906ca8968d1447871b77932f15ff1f8cfa53f05/src/bin/serde_from_map.rs

**Review by xxchan — COMMENTED** [review]
test

**Pull request closed without merging**